src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -699,7 +710,12 @@ export class PredictionTimeline {
 	 * Cursor position -- kept outside the buffer since it can be ahead if
 	 * typing swiftly.
 	 */
-	private cursor: Cursor | undefined;
+	private _physicalCursor: Cursor | undefined;
```

Comment:
"physical" here meaning "actual"? In otherwords where the cursor actually is verified by the shell?

Reply:
That's right. The position of the cursor that the user is currently looking at on their screen (or will be looking at after all pending writes are flushed)

Reply:
Clarified in docblocks